json-schema-generator/src/main/java/io/micronaut/jsonschema/generator/aggregator/TypeAggregator.java

```diff
@@ -0,0 +1,206 @@
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static io.micronaut.core.util.StringUtils.capitalize;
+import static io.micronaut.jsonschema.generator.aggregator.DefinitionsAggregator.getDefinitionType;
+import static java.lang.String.join;
+
+/**
+ * An aggregator for deducing type information from json schema.
+ *
+ * @author Elif Kurtay
+ * @since 1.2
+ */
+@Internal
+public class TypeAggregator {
+
+    private static final Map<String, TypeDef> TYPE_MAP = CollectionUtils.mapOf(new Object[]{
+        "integer", TypeDef.Primitive.INT, "boolean", TypeDef.Primitive.BOOLEAN, "array", TypeDef.of(List.class),
+        "void", TypeDef.VOID, "string", TypeDef.STRING, "object", TypeDef.OBJECT,
+        "number", TypeDef.Primitive.FLOAT, "null", TypeDef.OBJECT});
+
+    public static TypeDef getTypeDefFromJson(Map<String, Object> description) {
+        var type = description.getOrDefault("type", "object");
+        String typeName;
+        if (type.getClass() == ArrayList.class) {
+            typeName = ((ArrayList<?>) type).get(0).toString();
+        } else {
+            typeName = type.toString();
+        }
+
+        TypeDef typeDef;
+        if (typeName.equals("string") && description.containsKey("format")) {
+            var format = description.get("format").toString();
+            switch (format) {
+                case "date": typeDef = ClassTypeDef.of(LocalDate.class); break;
+                case "date-time", "time": typeDef = ClassTypeDef.of(ZonedDateTime.class); break;
+                case "duration": typeDef = ClassTypeDef.of(Duration.class); break;
+                case "ipv4": typeDef = ClassTypeDef.of(java.net.Inet4Address.class); break;
+                case "ipv6": typeDef = ClassTypeDef.of(java.net.Inet6Address.class); break;
+                case "uuid": typeDef = ClassTypeDef.of(UUID.class); break;
+                case "uri", "iri": typeDef = ClassTypeDef.of(URI.class); break;
+                case "json-pointer": typeDef = ClassTypeDef.of(JsonPointer.class); break;
+                // missing: email, web hostname, uri-reference, uri-template, regex
+                default: typeDef = TypeDef.STRING;
+            }
+        } else if (description.containsKey("$ref")) {
+            typeDef = getDefinitionType(description.get("$ref").toString());
+        } else {
+            typeDef = TYPE_MAP.get(typeName);
+        }
+        return typeDef;
+    }
+
+    public static Map<String, ?> getJsonSchema(InputStream inputStream, File schemaFile) throws IOException {
+        JsonMapper jsonMapper = new JsonMapper();
+        if (inputStream != null) {
+            String jsonString = new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
+            return (Map<String, ?>) jsonMapper.readValue(jsonString, HashMap.class);
+        } else if (schemaFile != null) {
+            return (Map<String, ?>) jsonMapper.readValue(schemaFile, HashMap.class);
+        }
+        return null;
```

Comment:
Instead of reading to a `Map`, can you try to use `Schema` as it is done in `json-schema-processor`.
I have created a PR to move it to a common module: https://github.com/micronaut-projects/micronaut-json-schema/pull/102.